lib/mayaUsd/commands/layerEditorCommand.cpp

```diff
@@ -694,15 +729,23 @@ MStatus LayerEditorCommand::parseArgs(const MArgList& argList)
     _layerIdentifier = objects[0].asChar();
 
     if (!isQuery()) {
+        Impl::IndexAdjustments indexAdjustments;
+
         if (argParser.isFlagSet(kInsertSubPathFlag)) {
             auto count = argParser.numberOfFlagUses(kInsertSubPathFlag);
             for (unsigned i = 0; i < count; i++) {
                 auto cmd = std::make_shared<Impl::InsertSubPath>();
 
                 MArgList listOfArgs;
                 argParser.getFlagArgumentList(kInsertSubPathFlag, i, listOfArgs);
-                cmd->_index = listOfArgs.asInt(0);
+
+                const int originalIndex = listOfArgs.asInt(0);
+                const int adjustedIndex = indexAdjustments.getAdjustedIndex(originalIndex);
```

Comment:
The methods on IndexAdjustments are fine for what they do, but the client always needs to call two methods, getAdjustedIndex(), then addInsertionAdjustment() or addRemovalAdjustment(), and both have the same argument (originalIndex).  Wouldn't it be nice to add a couple of additional methods to make the client's job less error-prone and simpler:

auto originalIndex = listOfAgs.asInt(0);
auto adjustedIndex = indexAdjustments.insertionAdjustment(originalIndex);

?  And similarly for removalAdjustment().  These two new convenience methods would simply call getAdjustedIndex() and addInsertionAdjustment() / addRemovalAdjustment().